amazon/ion/ioncmodule.c

```diff
@@ -1057,22 +1055,11 @@ iERR ionc_read_value(hREADER hreader, ION_TYPE t, PyObject* container, BOOL in_s
     PyObject*   py_annotations = NULL;
     PyObject*   py_value = NULL;
     PyObject*   ion_nature_constructor = NULL;
-
-    char    field_name_value[FIELD_NAME_MAX_LEN];
-    int     field_name_len = 0;
-    BOOL    None_field_name = TRUE;
+    PyObject*   py_field_name = NULL;
 
     if (in_struct) {
         IONCHECK(ion_reader_get_field_name(hreader, &field_name));
-        field_name_len = field_name.length;
-        if (field_name_len > FIELD_NAME_MAX_LEN) {
-            _FAILWITHMSG(IERR_INVALID_ARG,
-                "Filed name overflow, please try again with pure python.");
-        }
-        if (field_name.value != NULL) {
-            None_field_name = FALSE;
-            strcpy(field_name_value, field_name.value);
```

Comment:
Surprised this wasn't an issue previously, ion-c documents ION_STRINGs as not being null-terminated.

Reply:
Hah. It apparently was!
```
>>> from amazon.ion import simpleion
>>> simpleion.loads("{'f\0o': bar}")
[1]    18140 segmentation fault  python3
```

Works Now:
```
>>> from amazon.ion import simpleion
>>> simpleion.loads("{'f\0o': bar}")
{'f\x00o': IonPySymbol(text='bar', sid=None, location=None)}
```

Reply:
Created https://github.com/amazon-ion/ion-tests/pull/84 to add a test for this.